src/windows.rs

```diff
@@ -0,0 +1,210 @@
+//!
+//!  Windows-specific escaping.
+use std::borrow::Cow;
+use std::ffi::{OsStr, OsString};
+use std::iter::repeat;
+use std::os::windows::ffi::{OsStrExt, OsStringExt};
+
+/// Escape for the windows cmd.exe shell.
+///
+/// See [here][msdn] for more information.
+///
+/// [msdn]: http://blogs.msdn.com/b/twistylittlepassagesallalike/archive/2011/04/23/everyone-quotes-arguments-the-wrong-way.aspx
+pub fn escape(s: Cow<str>) -> Cow<str> {
+    let needs_escape = s.chars().any(|c| matches!(c, '"' | '\t' | '\n' | ' '));
+    if s.is_empty() || !needs_escape {
```

Comment:
I'm a little bit confused on why we should for `s.is_empty() || !needs_escape` here in windows, but check for `!s.is_empty() && needs_escape` on unix.

Reply:
I think they should be functionally equivalent but I've refactored them to look pretty similar now. lmk what you think.